app/path.go

```diff
@@ -228,18 +231,49 @@ func collectSchemaNodes(e *yang.Entry, leafOnly bool) []*yang.Entry {
 		collected = append(collected,
 			collectSchemaNodes(child, leafOnly)...)
 	}
+
 	if e.Parent != nil {
 		switch {
 		case e.Dir == nil && e.ListAttr != nil: // leaf-list
 			fallthrough
 		case e.Dir == nil: // leaf
-			collected = append(collected, e)
+			f := &yang.Entry{
+				Parent:      e.Parent,
+				Node:        e.Node,
+				Name:        e.Name,
+				Description: e.Description,
+				Default:     e.Default,
+				Units:       e.Units,
+				Kind:        e.Kind,
+				Config:      e.Config,
+				Prefix:      e.Prefix,
+				Mandatory:   e.Mandatory,
+				Dir:         e.Dir,
+				Key:         e.Key,
+				Type:        e.Type,
+				Exts:        e.Exts,
+				ListAttr:    e.ListAttr,
+				Extra:       make(map[string][]interface{}),
+			}
+			for k, v := range e.Extra {
+				f.Extra[k] = v
+			}
+			collected = append(collected, f)
 		case e.ListAttr != nil: // list
 			fallthrough
 		default: // container
 			if !leafOnly {
```

Comment:
The leafOnly case isn't clean with this PR : when running gnmic path with the --with-non-leaves, the conditions are not associated properly with each container path (in short, there is duplication in the FeatureList). Has to be corrected, or if-feature could be added as an optional flag, mutually exclusive with --with-non-leaves. What do you think @hellt @karimra ?

Reply:
Can we do it as we discussed on the call? Finish the collection then resolve the features recursively?